pkg/controller/nodeipam/ipam/cloud_cidr_allocator_metrics.go

```diff
@@ -0,0 +1,32 @@
+package ipam
+
+import (
+	"sync"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+// nodeIpamSubsystem - subsystem name used for Node IPAM Controller
+const nodeIpamSubsystem = "node_ipam_controller"
+
+var (
+	multiNetworkNodes = metrics.NewGaugeVec(
+		&metrics.GaugeOpts{
+			Subsystem:      nodeIpamSubsystem,
+			Name:           "multinetwork_node_total",
+			Help:           "Gauge measuring number of multinetworking nodes",
```

Comment:
is this metric useful? you can get this information from the node objects at anytime too

Reply:
Yes you can always see this information if you log into the controller and get the current state. This metric is useful for debugging and RCA. We want to see historical data on how many nodes were subscribed to a given network. The customer says something like "my pods were not being scheduled for a long time and then things started working after I upgraded a node pool".

Reply:
we should add comments to explain these things.
The name and help of this metrics can benefit of adding more details too